src/client.ts

```diff
@@ -105,7 +105,24 @@ export class Client {
   async queryAsync(contract: Address, query?: Message): Promise<Uint8Array | void> {
     const result = await this._readClient.sendAsync<string>('query', {
       contract: contract.local.toString(),
-      query: query ? Uint8ArrayToB64(query.serializeBinary()) : undefined
+      query: query ? Uint8ArrayToB64(query.serializeBinary()) : undefined,
+      vmType: VMType.PLUGIN
+    })
+    if (result) {
+      return B64ToUint8Array(result)
+    }
+  }
+
+  /**
+   * Queries the current state of a contract deployed on an EVM.
+   *
+   * Consider using Contract.staticCallAsync() instead.
+   */
+  async queryAsyncEVM(contract: Address, query?: Uint8Array): Promise<Uint8Array | void> {
```

Comment:
The only difference between `queryAsyncEVM` and `queryAsync` is the hard-coded vm type, can you just add an optional parameter to `queryAsync` instead of adding another method?

Reply:
Like so (default will be VMType.PLUGIN)
```typescript
 async queryAsync(contract: Address, query?: Message, vmType: VMType = VMType.PLUGIN): Promise<Uint8Array | void> {
     const result = await this._readClient.sendAsync<string>('query', {
       contract: contract.local.toString(),
       query: query ? Uint8ArrayToB64(query.serializeBinary()) : undefined,
       vmType,
    })
```

Reply:
Oh, I see. In that case lets move the message serialization to `Contract` and change `queryAsync` to:
```
async queryAsync(contract: Address, query?: Uint8Array, vmType: VMType = VMType.PLUGIN): Promise<Uint8Array | void> {
     const result = await this._readClient.sendAsync<string>('query', {
       contract: contract.local.toString(),
       query: query ? Uint8ArrayToB64(query) : undefined,
       vmType,
    })
```